src/lib.rs

```diff
@@ -25,6 +26,9 @@ pub trait StructOpt {
     /// Returns the corresponding `clap::App`.
     fn clap<'a, 'b>() -> clap::App<'a, 'b>;
 
+    /// Add this app's arguments/subcommands to another `clap::App`.
+    fn augment_clap<'a, 'b>(clap::App<'a, 'b>) -> clap::App<'a, 'b>;
```

Comment:
I could move that function into a inherent `impl` block for each type; that would remove it from cluttering up the `StructOpt` documentation, at least, which I think is the real issue.